src/pydargs/__init__.py

```diff
@@ -22,14 +36,45 @@ def parse(tp: Type[Dataclass], args: Optional[list[str]] = None) -> Dataclass:
 def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
     parser = ArgumentParser()
     for field in fields(tp):
-        parser.add_argument(
-            f"--{field.name.replace('_', '-')}",
-            default=field.default,
-            dest=field.name,
-            help=f"Override field {field.name}.",
-            type=field.type,
-        )
+        if origin := get_origin(field.type):
+            if origin is Sequence or origin is list:
+                if field.default is MISSING and field.default_factory is MISSING:
+                    raise NotImplementedError(f"Parsing {origin} without a default is not supported.")
+                parser.add_argument(
+                    f"--{field.name.replace('_', '-')}",
+                    default=argparse.SUPPRESS,
```

Comment:
```suggestion
                    default=argparse.SUPPRESS,  # Enforce this doesn't appear in the namespace when 0 args are provided
```

doesn't fit but something like this could be nice to add